server/src/main/java/org/apache/druid/segment/realtime/appenderator/BaseAppenderatorDriver.java

```diff
@@ -551,10 +551,19 @@ protected AppenderatorDriverAddResult append(
   )
   {
     if (segmentsAndMetadata.getSegments().isEmpty()) {
-      log.debug("Nothing to publish, skipping publish step.");
-      final SettableFuture<SegmentsAndMetadata> retVal = SettableFuture.create();
-      retVal.set(segmentsAndMetadata);
-      return retVal;
+      if (!publisher.supportsEmptyPublish()) {
+        log.info("Nothing to publish, skipping publish step.");
+        final SettableFuture<SegmentsAndMetadata> retVal = SettableFuture.create();
+        retVal.set(segmentsAndMetadata);
+        return retVal;
+      } else {
+        if (appenderator.getTotalRowCount() != 0) {
```

Comment:
could you leave a comment explaining the purpose of this check being in the else case, and explain why it isn't always an error?

Reply:
Added a comment there. I think the check could make sense across all task types, but would need more testing, so for this test I'm just limiting it to the seekable stream tasks